src/go/util/marshal_test.go

```diff
@@ -43,28 +40,36 @@ func TestUnmarshalBytesToPbMessage(t *testing.T) {
 		}
 	}
 
-	wantResp = &confpb.Service{
-		Id: "test-id",
-	}
-	getResp = &confpb.Service{}
-	_test()
-
-	wantResp = &smpb.ListServiceRolloutsResponse{
-		NextPageToken: "next-page-token",
-	}
-	getResp = &smpb.ListServiceRolloutsResponse{}
-	_test()
-
-	wantResp = &scpb.ReportResponse{
-		ServiceConfigId: "test-id",
+	testCases := []struct {
+		getResp   proto.Message
+		wantResp  proto.Message
+		wantError string
+	}{
+		{
+			getResp: &confpb.Service{},
```

Comment:
Can you add `desc` variable with string description? I still don't understand why there are fields populated in the `wantResp`. Like why is `Id` set in this test case, where did `test-id` come from?

Reply:
It is just to differentiate the gotMessage is right. Rename it to be gotMessageHolder.

Reply:
Ah i see